### PR TITLE
Add HuggingFace Model Wrapper for Semantic-Segmentation

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

image
*pipeline: /_base_/data/semantic_segmentation.yaml & optimizer: AdamW (lr 0.0001)
*The current settings are probably not optimal for each model, so we should be able to improve performance based on the more optimal settings.

### How to test

Install transformers
`pip install transformers`
or
`pip install -e .[transformers]`

API Examples
```python
from otx.algo.segmentation.huggingface_model import HuggingFaceModelForSegmentation
from otx.engine import Engine

data_root = "/datasets/otx_v2_dataset/semantic_seg/kvasir_medium"

otx_model = HuggingFaceModelForSegmentation(
    model_name_or_path="nvidia/segformer-b0-finetuned-ade-512-512",
    label_info=2,
)

engine = Engine(
    data_root=data_root,
    model=otx_model,
    work_dir="otx-workspace-hf",
)

engine.train(max_epochs=2)
engine.test()
engine.export()
engine.test(checkpoint="otx-workspace-hf/exported_model.xml")
```

CLI Examples
```shell
(venv) ~/otx$ otx train \
    --model otx.algo.segmentation.huggingface_model.HuggingFaceModelForSegmentation \
    --model.model_name_or_path nvidia/segformer-b0-finetuned-ade-512-512 \
    --data_root /datasets/otx_v2_dataset/semantic_seg/kvasir_medium \
    --work_dir otx-workspace-hf-cli --max_epochs 3
```

Unit-Test:
image

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/algo/segmentation/huggingface_model.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        return OTXNativeModelExporter(
+            task_level_export_parameters=self._export_parameters,
+            input_size=image_size,
+            mean=image_mean,
+            std=image_std,
+            resize_mode="standard",
+            pad_value=0,
+            swap_rgb=False,
+            via_onnx=False,
+            onnx_export_configuration=None,
```
Comment (kprokofi) [inline src/otx/algo/segmentation/huggingface_model.py:139]: Is it OK to leave ONNX config empty? Does it work?

**Review by kprokofi — APPROVED** [review]
Thank you for your work

**Review by wonjuleee — APPROVED** [review]
LGTM!

File: src/otx/algo/segmentation/huggingface_model.py (hunk 1 above)
Comment (harimkang) [reply src/otx/algo/segmentation/huggingface_model.py:139]: I haven't noticed any particular issues when I've tried it.

**Pull request merged**